aptos-move/framework/aptos-framework/sources/transaction_fee.move

```diff
@@ -1,15 +1,104 @@
+/// This module provides an interface to burn or collect and redistribute transaction fees.
 module aptos_framework::transaction_fee {
-    use aptos_framework::coin::{Self, BurnCapability};
+    use aptos_framework::coin::{Self, AggregatableCoin, BurnCapability, Coin};
     use aptos_framework::aptos_coin::AptosCoin;
+    use aptos_framework::stake;
     use aptos_framework::system_addresses;
+    use std::error;
+    use std::option::{Self, Option};
 
+    friend aptos_framework::block;
     friend aptos_framework::genesis;
+    friend aptos_framework::reconfiguration;
     friend aptos_framework::transaction_validation;
 
+    /// When gas fees are already being collected and the struct holding
+    /// information about collected amounts is already published.
+    const EALREADY_COLLECTING_FEES: u64 = 1;
+
+    /// When the burn percentage is out of range [0, 100].
+    const EINVALID_BURN_PERCENTAGE: u64 = 2;
+
+    /// Stores burn capability to burn the gas fees.
     struct AptosCoinCapabilities has key {
         burn_cap: BurnCapability<AptosCoin>,
     }
 
+    /// Stores information about the block proposer and the amount of fees
+    /// collected when executing the block.
+    struct CollectedFeesPerBlock has key {
+        amount: AggregatableCoin<AptosCoin>,
+        proposer: Option<address>,
+        burn_percentage: u8,
+    }
+
+    /// Initializes the resource storing information about gas fees collection and
+    /// distribution. Should be called by on-chain governance.
+    public fun initialize_fee_collection_and_distribution(aptos_framework: &signer, burn_percentage: u8) {
+        // Sanity checks.
+        system_addresses::assert_aptos_framework(aptos_framework);
+        assert!(
+            !exists<CollectedFeesPerBlock>(@aptos_framework),
+            error::already_exists(EALREADY_COLLECTING_FEES)
+        );
+        assert!(burn_percentage <= 100, error::out_of_range(EINVALID_BURN_PERCENTAGE));
+
+        // Make sure stakng module is aware of transaction fees collection.
+        stake::initialize_validator_fees(aptos_framework);
+
+        // Initially, no fees are collected and the block proposer is not set.
+        let zero = coin::initialize_aggregatable_coin(aptos_framework);
```

Comment:
nit: Why not just inline coin::initialize_aggregatable_coin(aptos_framework); below?